Unity/Assets/NativeScript/Bindings.cs

```diff
@@ -24,44 +25,36 @@ public static class Bindings
 		// Holds objects and provides handles to them in the form of ints
 		public static class ObjectStore
 		{
+			// Lookup handles by object.
+			static Dictionary<object, int> objectHandleCache;
+
+			// Stack of available handles.
+			static Stack<int> handles;
+			
 			// Stored objects. The first is never used so 0 can be "null".
 			static object[] objects;
 			
-			// Stack of available handles
-			static int[] handles;
-			
-			// Hash table of stored objects to their handles.
-			static object[] keys;
-			static int[] values;
-			
-			// Index of the next available handle
-			static int nextHandleIndex;
-			
 			// The maximum number of objects to store. Must be positive.
 			static int maxObjects;
 			
 			public static void Init(int maxObjects)
 			{
 				ObjectStore.maxObjects = maxObjects;
+				objectHandleCache = new Dictionary<object, int> (maxObjects);	
```

Comment:
For consistency, please remove the space between the type name and the parameters:
```csharp
new Dictionary<object, int>(maxObjects)
```
(same on the next line)